### PR TITLE
feat: gatsby-ify

### PR DESCRIPTION
**Pull request by cdaringe**
# problemo

- lots of window hacks in this beautiful, simple, fast jade impl

# solution

- proper compile phase for static site via gatsby... which also has a few hacks up its sleeve

### CODE REVIEW

File: packages/effects-docs/.gitignore [hunk 1]
```diff
@@ -1,14 +1,83 @@
...
 node_modules
 .DS_Store
-effectsjs/highlight/**/*css
-effectsjs/highlight/**/*jpg
-effectsjs/highlight/**/*png
-effectsjs/highlight/**/*md
+src/highlight/**/*css
+src/highlight/**/*jpg
+src/highlight/**/*png
+src/highlight/**/*md
```
Comment (cdaringe) [inline packages/effects-docs/.gitignore]: pruning. should no longer be needed

File: packages/effects-docs/.gitignore [hunk 2]
```diff
@@ -1,14 +1,83 @@
...
 babel.js
 
 # injected during build
 *.bundle.js
+
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (cdaringe) [inline packages/effects-docs/.gitignore]: why was *.log insufficient? pruning.

File: .eslintrc.js [hunk 3]
```diff
@@ -1,21 +1,41 @@
 module.exports = {
   ignorePatterns: [
-    "node_modules",
+    "coverage",
+    "dist",
+    "**/node_modules/**",
```
Comment (cdaringe) [inline .eslintrc.js:5]: was this necessary? 🤔

File: package.json [hunk 4]
```diff
@@ -4,9 +4,11 @@
   "description": "",
   "scripts": {
     "bootstrap": "npm install && npx lerna bootstrap --loglevel verbose",
+    "postbootstrap": "run-s copy:nvmrc",
+    "copy:nvmrc": "lerna exec -- cp ../../.nvmrc ./",
     "build": "tsc -b packages && lerna exec --scope effects-docs -- npm run build",
     "clean": "del 'packages/*/lib' 'packages/*/*.{tsbuildinfo}' 'packages/*/src/**/*.{map,d.ts}'",
-    "format:dryrun": "prettier ./*.{js,ts} ./packages/*/{test,src}/**/*.{js,ts}",
+    "format:dryrun": "prettier './*.{js,ts}' './packages/**/*.{js,ts}'",
```
Comment (cdaringe) [inline package.json:11]: previously was using shell expansion. whoops

File: packages/effects-docs/gatsby-node.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ */
+const { createFilePath } = require(`gatsby-source-filesystem`);
+const { toExamples, withoutExampleSuffix } = require("./src/util/examples.js");
+const fs = require("fs");
+const path = require("path");
+const examplesGql = `
+{
+  examples: allFile(filter: {
+    relativePath: {
+      regex: "/examples/.*.example$/"
```
Comment (cdaringe) [inline packages/effects-docs/gatsby-node.js:14]: even if something _isnt required_--gatsby tries to parse it.  v. obnoxious--forced me to write. `x.js` files as `x.js.example`

Comment (cdaringe) [reply packages/effects-docs/gatsby-node.js:14]: ...due to effects.  i _guess_ i could wire gatsby's babel to _our_ babel 🤔 . another time.

File: packages/effects-docs/scripts/reduce-highlight.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      })
+      .join("\n") + "\n";
+  return writeFile(highlightIndex, nextContents, "utf8");
+  console.log(
+    `[reduce-highlight.js]: highlight languages reduced to ${whitelist.join(
+      ", "
+    )}`
+  );
+}
+go();
```
Comment (cdaringe) [inline packages/effects-docs/scripts/reduce-highlight.js:26]: without this, highlight.js is quite expensive.  turn off other language packs!

File: packages/effects-docs/src/components/editor.js [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    level === "warn" ||
+    level === "error" ||
+    level === "log";
+  const withTap = (fn, fn2) => (...args) => {
+    fn(...args);
+    return fn2(...args);
+  };
+  const appendLog = (level, ...args) => {
+    // @warn react-events induce various non-controlled patched log calls on unmount,
+    // so filter during
```
Comment (cdaringe) [inline packages/effects-docs/src/components/editor.js]: incomplete sentence

File: packages/effects-docs/src/util/window-location.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+function getWindow() {
```
Comment (cdaringe) [inline packages/effects-docs/src/util/window-location.js:4]: for ssr gatsby stuff. will label

File: packages/effects-docs/src/components/editor.js [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+    const AceEditor = this.AceEditor || (() => <h3>Downloading editor...</h3>);
+    return (
+      <div id="editor_container" className={isFullScreen ? "fullscreen" : ""}>
+        <Helmet>
+          {/**
+           * babel (standalone) makes webpack _lose its cool_ in production mode,
+           * which is too bad, because webpack should minify it.  it's so bad,
+           * that we're ejecting to script loading this little bugger
+           * */}
+          <script src="/effectsjs/babel.js"></script>
```
Comment (dpchamps) [inline packages/effects-docs/src/components/editor.js]:
```suggestion
         <script src={process.env.NODE_ENV === 'development' ? '/babel.js' : '/effectsjs/babel.js'}></script>
```

lame i know

**Review by dpchamps — CHANGES_REQUESTED** [review]

**Review by dpchamps — APPROVED** [review]
💪💪💪💪💪

File: packages/effects-docs/src/pages/examples.js [hunk 10]
```diff
@@ -38,7 +38,7 @@ const Examples = () => {
               <td>{i + 1}</td>
               <td>
                 <Link
-                  to={`examples/${withoutExampleSuffix(
+                  to={`/examples/${withoutExampleSuffix(
```
Comment (cdaringe) [inline packages/effects-docs/src/pages/examples.js]: AND THIS lack of `/` was what was causing the warning message from gatsby's dev runtime.

**Pull request merged**